language/move-lang/src/cfgir/ast.rs

```diff
@@ -298,22 +298,37 @@ impl UnannotatedExp_ {
 }
 
 impl BaseType_ {
-    pub fn builtin(loc: Loc, b_: BuiltinTypeName_) -> BaseType {
```

Comment:
Make this take type arguments

Reply:
Then you can replicate logic around inferring the kind. Case on the kind provided by `b_.kind()`, if its unrestricted. Infer the kind from the type arguments.